### ISSUE TITLE
TASK [install node packages] FAILED

### ISSUE THREAD

**Issue by eelcodevlieger** [openfoodfoundation/ofn-devbox#1]
I am seeing an error during the provisioning stage for task name 'install node packages'.

> "**stderr**": "npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to \"coffeescript\" (no hyphen)\nnpm

RAW output of failed task below:
```
TASK [install node packages] ***************************************************
fatal: [ofn-devbox]: FAILED! => {"changed": true, "cmd": "bash -lc \"npm install\"", "delta": "0:00:39.322111", "end": "2019-04-27 13:56:49.195491", "msg": "non-zero return code", "rc": 185, "start": "2019-04-27 13:56:09.873380", "stderr": "npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to \"coffeescript\" (no hyphen)\nnpm WARN optional Skipping failed optional dependency /chokidar/fsevents:\nnpm WARN notsup Not compatible with your operating system or architecture: fsevents@1.2.8\nnpm ERR! Linux 4.4.0-146-generic\nnpm ERR! argv \"/home/vagrant/.nodenv/versions/5.12.0/bin/node\" \"/home/vagrant/.nodenv/versions/5.12.0/bin/npm\" \"install\"\nnpm ERR! node v5.12.0\nnpm ERR! npm  v3.8.6\nnpm ERR! path ../atob/bin/atob.js\nnpm ERR! code EPROTO\nnpm ERR! errno -71\nnpm ERR! syscall symlink\n\nnpm ERR! EPROTO: protocol error, symlink '../atob/bin/atob.js' -> '/home/vagrant/openfoodnetwork/node_modules/.bin/atob'\nnpm ERR! \nnpm ERR! If you need help, you may report this error at:\nnpm ERR!     <https://github.com/npm/npm/issues>\n\nnpm ERR! Please include the following file with any support request:\nnpm ERR!     /home/vagrant/openfoodnetwork/npm-debug.log", "stderr_lines": ["npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to \"coffeescript\" (no hyphen)", "npm WARN optional Skipping failed optional dependency /chokidar/fsevents:", "npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.2.8", "npm ERR! Linux 4.4.0-146-generic", "npm ERR! argv \"/home/vagrant/.nodenv/versions/5.12.0/bin/node\" \"/home/vagrant/.nodenv/versions/5.12.0/bin/npm\" \"install\"", "npm ERR! node v5.12.0", "npm ERR! npm  v3.8.6", "npm ERR! path ../atob/bin/atob.js", "npm ERR! code EPROTO", "npm ERR! errno -71", "npm ERR! syscall symlink", "", "npm ERR! EPROTO: protocol error, symlink '../atob/bin/atob.js' -> '/home/vagrant/openfoodnetwork/node_modules/.bin/atob'", "npm ERR! ", "npm ERR! If you need help, you may report this error at:", "npm ERR!     <https://github.com/npm/npm/issues>", "", "npm ERR! Please include the following file with any support request:", "npm ERR!     /home/vagrant/openfoodnetwork/npm-debug.log"], "stdout": "openfoodnetwork@1.7.1 /home/vagrant/openfoodnetwork\n├── anymatch@1.3.2  extraneous\n├── async-each@1.0.3  extraneous\n├── base@0.11.2  extraneous\n├── batch@0.5.3  extraneous\n├── bluebird@2.11.0  extraneous\n├── body-parser@1.19.0  extraneous\n├── colors@1.3.3  extraneous\n├── connect@3.6.6  extraneous\n├── core-js@2.6.5  extraneous\n├── decode-uri-component@0.2.0  extraneous\n├── di@0.0.1  extraneous\n├── dom-serialize@2.2.1  extraneous\n├── expand-braces@0.1.2  extraneous\n├── fragment-cache@0.2.1  extraneous\n├── graceful-fs@4.1.15  extraneous\n├── http-proxy@1.17.0  extraneous\n├── is-binary-path@1.0.1  extraneous\n├── is-windows@1.0.2  extraneous\n├── isbinaryfile@3.0.3  extraneous\n├── jasmine-core@2.4.1 \n├── UNMET PEER DEPENDENCY karma@>=0.11.14\n├─┬ karma-chrome-launcher@2.2.0 \n│ ├─┬ fs-access@1.0.1 \n│ │ └── null-check@1.0.0 \n│ └─┬ which@1.3.1 \n│   └── isexe@2.0.0 \n├─┬ karma-coffee-preprocessor@0.3.0 \n│ └── coffee-script@1.12.7 \n├── karma-jasmine@0.3.8 \n├── lodash@3.10.1  extraneous\n├── log4js@0.6.38  extraneous\n├── mime@1.6.0  extraneous\n├── object.pick@1.3.0  extraneous\n├── optimist@0.6.1  extraneous\n├── posix-character-classes@0.1.1  extraneous\n├── readable-stream@2.3.6  extraneous\n├── resolve-url@0.2.1  extraneous\n├── rimraf@2.6.3  extraneous\n├── snapdragon-node@2.1.1  extraneous\n├── socket.io@1.7.4  extraneous\n├── source-map@0.5.7  extraneous\n├── source-map-url@0.4.0  extraneous\n├── to-regex@3.0.2  extraneous\n├── to-regex-range@2.1.1  extraneous\n├── urix@0.1.0  extraneous\n├── use@3.1.1  extraneous\n└── useragent@2.3.0  extraneous", "stdout_lines": ["openfoodnetwork@1.7.1 /home/vagrant/openfoodnetwork", "├── anymatch@1.3.2  extraneous", "├── async-each@1.0.3  extraneous", "├── base@0.11.2  extraneous", "├── batch@0.5.3  extraneous", "├── bluebird@2.11.0  extraneous", "├── body-parser@1.19.0  extraneous", "├── colors@1.3.3  extraneous", "├── connect@3.6.6  extraneous", "├── core-js@2.6.5  extraneous", "├── decode-uri-component@0.2.0  extraneous", "├── di@0.0.1  extraneous", "├── dom-serialize@2.2.1  extraneous", "├── expand-braces@0.1.2  extraneous", "├── fragment-cache@0.2.1  extraneous", "├── graceful-fs@4.1.15  extraneous", "├── http-proxy@1.17.0  extraneous", "├── is-binary-path@1.0.1  extraneous", "├── is-windows@1.0.2  extraneous", "├── isbinaryfile@3.0.3  extraneous", "├── jasmine-core@2.4.1 ", "├── UNMET PEER DEPENDENCY karma@>=0.11.14", "├─┬ karma-chrome-launcher@2.2.0 ", "│ ├─┬ fs-access@1.0.1 ", "│ │ └── null-check@1.0.0 ", "│ └─┬ which@1.3.1 ", "│   └── isexe@2.0.0 ", "├─┬ karma-coffee-preprocessor@0.3.0 ", "│ └── coffee-script@1.12.7 ", "├── karma-jasmine@0.3.8 ", "├── lodash@3.10.1  extraneous", "├── log4js@0.6.38  extraneous", "├── mime@1.6.0  extraneous", "├── object.pick@1.3.0  extraneous", "├── optimist@0.6.1  extraneous", "├── posix-character-classes@0.1.1  extraneous", "├── readable-stream@2.3.6  extraneous", "├── resolve-url@0.2.1  extraneous", "├── rimraf@2.6.3  extraneous", "├── snapdragon-node@2.1.1  extraneous", "├── socket.io@1.7.4  extraneous", "├── source-map@0.5.7  extraneous", "├── source-map-url@0.4.0  extraneous", "├── to-regex@3.0.2  extraneous", "├── to-regex-range@2.1.1  extraneous", "├── urix@0.1.0  extraneous", "├── use@3.1.1  extraneous", "└── useragent@2.3.0  extraneous"]}
	to retry, use: --limit @/vagrant/setup/provision.retry

PLAY RECAP *********************************************************************
ofn-devbox                 : ok=27   changed=8    unreachable=0    failed=1   
```

**Comment by lin-d-hop** [comment]
Hi @eelcodevlieger!
Welcome!
@Matt-Yorkley can you take a look at these issues?

**Comment by eelcodevlieger** [comment]
Thank you!

The openfoodnetwork\npm-debug.log also shows NPM problems.

> EPROTO: protocol error, **symlink**

Could be related to this issue:
https://github.com/npm/npm/issues/992

Raw log errors:
```
29952 warn optional Skipping failed optional dependency /chokidar/fsevents:
29953 warn notsup Not compatible with your operating system or architecture: fsevents@1.2.8
29954 verbose stack Error: EPROTO: protocol error, symlink '../atob/bin/atob.js' -> '/home/vagrant/openfoodnetwork/node_modules/.bin/atob'
29954 verbose stack     at Error (native)
29955 verbose cwd /home/vagrant/openfoodnetwork
29956 error Linux 4.4.0-146-generic
29957 error argv "/home/vagrant/.nodenv/versions/5.12.0/bin/node" "/home/vagrant/.nodenv/versions/5.12.0/bin/npm" "install"
29958 error node v5.12.0
29959 error npm  v3.8.6
29960 error path ../atob/bin/atob.js
29961 error code EPROTO
29962 error errno -71
29963 error syscall symlink
29964 error **EPROTO: protocol error, symlink** '../atob/bin/atob.js' -> '/home/vagrant/openfoodnetwork/node_modules/.bin/atob'
29965 error If you need help, you may report this error at:
29965 error     <https://github.com/npm/npm/issues>
29966 verbose exit [ -71, true ]
```

**Comment by eelcodevlieger** [comment]
Looking at the error again, I think the coffee script part is just a warning and the main issue preventing me from starting the vagrant box is the NPM symlink issue.

It is trying to create a symlink under /home/vagrant/openfoodnetwork/, which is a VM synced folder.
```
symlink '../atob/bin/atob.js' -> '/home/vagrant/openfoodnetwork/node_modules/.bin/atob'"
```

From Vagrantfile:
```
  config.vm.synced_folder "openfoodnetwork/", "/home/vagrant/openfoodnetwork", id: "openfoodnetwork", create: true
```

Might be able to enable symlinks on Windows in combination with VirtualBox with the following command and running both VirtualBox and the vagrant commands as administrator.
```
  config.vm.provider :virtualbox do |vbox|
    vbox.name = "ofn-devbox"
    vbox.customize ["setextradata", "ofn-devbox", "VBoxInternal2/SharedFoldersEnableSymlinksCreate/openfoodnetwork", "1"]
```

This is the equivalent of executing command:
```
"c:\Program Files\Oracle\VirtualBox\VBoxManage.exe" setextradata "ofn-devbox" VBoxInternal2/SharedFoldersEnableSymlinksCreate/openfoodnetwork 1
```

Not had this working successfully yet though :(

Some links:
https://www.speich.net/articles/en/2018/12/24/virtualbox-6-how-to-enable-symlinks-in-a-linux-guest-os/
Alternative approach: https://blog.puphpet.com/blog/2015/06/25/windows-symlinks/

**Comment by eelcodevlieger** [comment]
I managed to get it to work! Now seeing the Open Food Network website via localhost.

Not sure what fixed it. Since my last comment I tried 2 things

1. run secpol.scm (local security policy). Opened Local Policies > User Rights Assignment > Create symbolic links. Added both my user and SYSTEM to the list (Administrators was already in there).
2. killed the full process tree of VirtualBox, including related services.

Then ran both VirtualBox and the vagrant commands as administrator.

If I get some time tomorrow I'll rerun this on another machine to narrow down the config and actions required.
